frontend/src/app/modules/fields/edit/field-types/te-work-package-edit-field.component.ts

```diff
@@ -28,11 +28,44 @@
 
 import {Component} from "@angular/core";
 import {WorkPackageEditFieldComponent} from "core-app/modules/fields/edit/field-types/work-package-edit-field.component";
+import {TimeEntryDmService} from "core-app/modules/hal/dm-services/time-entry-dm.service";
+import {ApiV3FilterBuilder} from "core-components/api/api-v3/api-v3-filter-builder";
+import {
+  TimeEntryWorkPackageAutocompleterComponent,
+  TimeEntryWorkPackageAutocompleterMode
+} from "core-app/modules/common/autocomplete/te-work-package-autocompleter.component";
+import {HalResource} from "core-app/modules/hal/resources/hal-resource";
+
+const RECENT_TIME_ENTRIES_MAGIC_NUMBER = 30;
 
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/opf/openproject/blob/641fb53a4498e565ea86c2645ff34bd3b879a0f5/frontend/src/app/modules/fields/edit/field-types/te-work-package-edit-field.component.ts#L40"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>